integration-tests/test/shared/env.ts

```diff
@@ -106,4 +122,82 @@ export class OptimismEnv {
   ): Promise<CrossDomainMessagePair> {
     return waitForXDomainTransaction(this.watcher, tx, direction)
   }
+
+  async relayXDomainMessages(
+    tx: Promise<TransactionResponse> | TransactionResponse
+  ): Promise<void> {
+    tx = await tx
+
+    let messagePairs = []
+    while (true) {
+      try {
+        messagePairs = await getMessagesAndProofsForL2Transaction(
+          l1Provider,
+          l2Provider,
+          this.scc.address,
+          predeploys.OVM_L2CrossDomainMessenger,
+          tx.hash
+        )
+        break
+      } catch (err) {
+        if (err.message.includes('unable to find state root batch for tx')) {
+          await sleep(5000)
+        } else {
+          throw err
+        }
+      }
+    }
+
+    for (const { message, proof } of messagePairs) {
+      while (true) {
+        try {
+          const result = await this.l1Messenger
+            .connect(this.l1Wallet)
+            .relayMessage(
+              message.target,
+              message.sender,
+              message.message,
+              message.messageNonce,
+              proof
+            )
+          await result.wait()
+          break
+        } catch (err) {
+          if (err.message.includes('execution failed due to an exception')) {
+            await sleep(5000)
+          } else if (
+            err.message.includes('message has already been received')
+          ) {
+            break
+          } else {
+            throw err
+          }
+        }
+      }
+    }
+  }
+}
+
+export const useDynamicTimeoutForWithdrawals = async (
```

Comment:
In the spirit of writing more testable code, our test-code should be better documented. Could you add a comment to this function and the previous? 
```suggestion
/**
 * Sets the timeout based on the challenge period of the network.
 * If the challenge period is greater than 60 seconds (ie. Mainnet), skip the test.
 */
export const useDynamicTimeoutForWithdrawals = async (
```